django/contrib/admin/helpers.py

```diff
@@ -438,6 +443,12 @@ def inline_formset_data(self):
     def forms(self):
         return self.formset.forms
 
+    @cached_property
+    def is_collapsible(self):
+        if any(self.formset.errors):
+            return False
+        return "collapse" in self.classes
+
```

Comment:
Is this and the changes to `django/contrib/admin/templates/admin/edit_inline/tabular.html` and `django/contrib/admin/templates/admin/edit_inline/stacked.html` required?
The way I read the existing feature it looks like it is applied only for fieldsets rather than formsets.

Reply:
Yes it is. Formsets can also be collapsible; you can put a collapsible fieldset inside a collapsible stacked or tabular inline.

The original logic for fieldsets is that they should not be collapsible when one of its fields has an error. Originally, JavaScript performed that check.

It makes sense to restrict complete inlines from collapsing in case of an error, since that is in line with the fieldset logic. This way we ensure that error'ed fields stay visible.

Reply:
@sarahboyce Forgot to tag you.

Reply:
I see, I missed that the inline html are using `<fieldset></fieldset>` 👍

Reply:
I appreciate this conversation because it addressed some questions that I had, but there is one issue with "automatically expanding a fieldset that has errors" in that, for example, a fieldset inside a `StackedInline` would be rendered as an `h4`, but when there are errors, it's rendered as an `h2`. This is also an issue in main (though it uses `h3` instead of `h4`) so I created this ticket-35456 to track this.